src/fides/api/service/connectors/query_config.py

```diff
@@ -337,8 +337,10 @@ def generate_update_stmt(
         return None
 
 
-class SQLQueryConfig(QueryConfig[Executable]):
-    """Query config that translates parameters into SQL statements."""
+class SQLLikeQueryConfig(QueryConfig[T], ABC):
+    """
+    Abstract query config for SQL-like languages (that may not be strictly SQL).
+    """
```

Comment:
Yes, and also `generate_update_stmt` and `format_fields_for_query`. 
For clarification: 

- Both `format_fields_for_query` and `get_update_stmt` keep their implementations with no changes, they were just moved up from `SQLQueryConfig` into `SQLLikeQueryConfig`. 

- Both `generate_query_without_tuples` and `generate_update_stmt`  were also part of `SQLQueryConfig` and I kept almost the entire implementation, only changing a few lines that were integration-specific, replacing them with calls to some of these other abstract methods that each class implements. 

- These integration-specific methods are `format_query_data_name`, `get_formatted_query_string`, `get_update_clauses`, `format_query_stmt`, `format_clause_for_query` and  `format_key_map_for_update_stmt`.  
    - Most of these are needed because Scylla uses a different syntax for expressing parameters in a query: SQL databases, that we connect to with SQLAlchemy , use `:` to denote a parameter , e.g `SELECT user_id FROM users WHERE email = :email`, while Scylla uses `%()s` , e.g `SELECT user_id FROM users WHERE email = %(email)s`.
    - Another difference is in the fact that for Scylla queries, we need to pass in the query parameters to the `execute` method (called in the connector itself), so we return the query and the query params separately. Meanwhile for SQL queries , we call the SQLAlchemy `text` function and pass in the parameters to it, so the query config returns the query with the parameters already replaced